### PR TITLE
Fix LRO test

### PR DESCRIPTION
**Pull request by harihare**
Currently LRO test does not run the LRO "on" codepath.
Patch will fix the indentation issue so that it takes
LRO "on" path.

Signed-off-by: Hariharan T.S <hari@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harihare** [comment]
# avocado run net_data.py:NetDataTest.testlro -m net_data.py.data/net_data.yaml 
JOB ID     : 4605d696d76c6b3ee3a765d5eab69511769ca64e
JOB LOG    : /home/hariharan/avocado-fvt-wrapper/results/job-2019-05-22T02.04-4605d69/job.log
 (1/1) net_data.py:NetDataTest.testlro;Interface1-MTU-ad26: ERROR: Command 'ethtool -K eth4 lro on' failed.\nstdout: ''\nstderr: 'Could not change any device features\n'\nadditional_info: None (13.92 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 17.67 s
JOB HTML   : /home/hariharan/avocado-fvt-wrapper/results/job-2019-05-22T02.04-4605d69/results.html


The failure is the actual  bug. Which was not hit with out the patch.

**Comment by harihare** [comment]
2019-05-22 02:04:16,531 net_data         L0174 INFO | Largest Receive Offload
2019-05-22 02:04:16,531 process          L0626 INFO | Running 'ethtool -K eth4 lro off'
2019-05-22 02:04:16,558 process          L0714 INFO | Command 'ethtool -K eth4 lro off' finished with 0 after 0.00289297103882s
2019-05-22 02:04:16,565 process          L0626 INFO | Running 'ping -c 1 152.1.1.119'
2019-05-22 02:04:16,752 process          L0458 DEBUG| [stdout] PING 152.1.1.119 (152.1.1.119) 56(84) bytes of data.
2019-05-22 02:04:16,758 process          L0714 INFO | Command 'ping -c 1 152.1.1.119' finished with 0 after 0.190588951111s
2019-05-22 02:04:16,810 process          L0458 DEBUG| [stdout] 64 bytes from 152.1.1.119: icmp_seq=1 ttl=64 time=0.072 ms
2019-05-22 02:04:16,810 process          L0458 DEBUG| [stdout]
2019-05-22 02:04:16,814 process          L0458 DEBUG| [stdout] --- 152.1.1.119 ping statistics ---
2019-05-22 02:04:16,814 process          L0458 DEBUG| [stdout] 1 packets transmitted, 1 received, 0% packet loss, time 0ms
2019-05-22 02:04:16,814 process          L0458 DEBUG| [stdout] rtt min/avg/max/mdev = 0.072/0.072/0.072/0.000 ms
2019-05-22 02:04:16,821 process          L0626 INFO | Running 'ethtool -K eth4 lro on'
2019-05-22 02:04:16,855 process          L0458 DEBUG| [stderr] Could not change any device features
2019-05-22 02:04:16,855 process          L0714 INFO | Command 'ethtool -K eth4 lro on' finished with 1 after 0.0100700855255s
2019-05-22 02:04:16,931 stacktrace       L0041 ERROR|
2019-05-22 02:04:16,991 stacktrace       L0044 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_framework-68.0-py2.7.egg/avocado/core/test.py:839
2019-05-22 02:04:17,035 stacktrace       L0047 ERROR| Traceback (most recent call last):
2019-05-22 02:04:17,035 stacktrace       L0047 ERROR|   File "/home/hariharan/avocado-fvt-wrapper/tests/avocado-misc-tests/io/net/net_data.py", line 182, in testlro
2019-05-22 02:04:17,036 stacktrace       L0047 ERROR|     if process.system(msg, shell=True) != 0:
2019-05-22 02:04:17,036 stacktrace       L0047 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_framework-68.0-py2.7.egg/avocado/utils/process.py", line 1446, in system
2019-05-22 02:04:17,036 stacktrace       L0047 ERROR|     encoding=encoding)
2019-05-22 02:04:17,036 stacktrace       L0047 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_framework-68.0-py2.7.egg/avocado/utils/process.py", line 1383, in run
2019-05-22 02:04:17,037 stacktrace       L0047 ERROR|     raise CmdError(cmd, sp.result)
2019-05-22 02:04:17,037 stacktrace       L0047 ERROR| CmdError: Command 'ethtool -K eth4 lro on' failed.
2019-05-22 02:04:17,037 stacktrace       L0047 ERROR| stdout: ''
2019-05-22 02:04:17,037 stacktrace       L0047 ERROR| stderr: 'Could not change any device features\n'
2019-05-22 02:04:17,037 stacktrace       L0047 ERROR| additional_info: None

File: io/net/net_data.py [hunk 1]
```diff
@@ -178,12 +178,12 @@ def testlro(self):
         ret = process.system("ping -c 1 %s" % self.peer, shell=True)
         if ret != 0:
             self.fail("lro test failed")
-            msg = "ethtool -K %s lro on" % self.iface
-            if process.system(msg, shell=True) != 0:
-                self.fail("LRO Test failed")
-                ret = process.system("ping -c 1 %s" % self.peer, shell=True)
-                if ret != 0:
-                    self.fail("lro test failed")
+        msg = "ethtool -K %s lro on" % self.iface
```
Comment (narasimhan-v) [inline io/net/net_data.py]: @harihare this can be done like:
```
for state in ['on', 'off']:
    lro state
    if fail:
        failed to state
```

Comment (narasimhan-v) [reply io/net/net_data.py]: @harihare while at it, can you also do the changes to gro test too ?

**Comment by narasimhan-v** [comment]
hope https://github.com/avocado-framework-tests/avocado-misc-tests/pull/1176 takes care of this in a different way

**Pull request closed without merging**